client/app/visualizations/vega/Renderer.jsx

```diff
@@ -0,0 +1,229 @@
+      try {
+        spec = vl.compile(spec).spec;
+      } catch (err) {
+        error = err.message;
+      }
+      // revert to origin height if we are doing autoresize
+      // (Vega-Lite will set default size as 200x200)
+      if (autoresize) {
+        spec.width = width;
+        spec.height = height;
+      }
+    }
+
+    return { error, mode, spec, autoresize };
+  });
+
+  constructor(props) {
+    super(props);
+    this.state = { width: 0, height: 0 };
+  }
+
+  componentDidMount() {
+    // eslint-disable-next-line compat/compat
+    this.resizeObserver = new ResizeObserver((entries) => {
+      const rect = entries[0].contentRect;
+      if (rect.width && rect.height) {
+        this.updateLayout({ parentSize: rect });
+      }
+    });
+    this.updateLayout();
+    this.resizeObserver.observe(this.elem.offsetParent);
+  }
+
+  componentWillUnmount() {
+    this.resizeObserver.disconnect();
+  }
+
+  /**
+   * Parse component.props
+   */
+  parseProps = ({ options, data }) => {
+    const { error, mode, spec, autoresize } = this.parseOptions(options);
+    const specData = spec.data && spec.data[0];
+    if (data && specData && specData.name === 'current_query') {
+      // Inject actual data to the data source in spec
+      specData.values = data.rows;
+      // ignore `url` and `format` config
+      delete specData.url;
+      delete specData.format;
+    }
+    if (!specData) {
+      spec.data = [{ values: data.rows }];
+    }
+    return { error, mode, spec, autoresize };
+  };
+
+  /**
+   * Updaete width & height in spec based on parent size
+   * @param {number} width - manual width in pixels
+   * @param {number} height - manual height in pixels
+   * @param {number} parentSize - parent width and height
+   */
+  updateLayout({ width, height, parentSize } = {}) {
+    // when there is error message or element is unmounting
+    // these elements might be null
+    if (!this.vega || !this.vega.element) return;
+    const { spec, autoresize } = this.parseOptions(this.props.options);
+    if (!spec || !autoresize) return;
+    if (!parentSize) {
+      const node = this.elem.offsetParent;
+      const bounds = node.getBoundingClientRect();
+      parentSize = {
+        width: bounds.width,
+        height: bounds.height,
+      };
+    }
+    const { width: specWidth, height: specHeight } = spec;
+    let hPadding = 20;
+    // if from editor, needs space for the edit link
+    let vPadding = this.props.fromEditor ? 40 : 5;
+    if (typeof spec.padding === 'number') {
+      hPadding += 2 * spec.padding;
+      vPadding += 2 * spec.padding;
+    } else if (isObject(spec.padding)) {
+      hPadding += (spec.padding.left || 0) + (spec.padding.right || 0);
+      vPadding += (spec.padding.top || 0) + (spec.padding.bottom || 0);
+    }
+    width = width || specWidth || Math.max(parentSize.width - hPadding, 100);
+    height = height || specHeight || Math.min(400, Math.max(parentSize.height - vPadding, 300));
```

Comment:
By default, a Vega visualization has min-height 300px and max-height 400px. This is inline with what Plot.ly currently has, although I was not able to find where we set the height for Plot.ly...

Reply:
We don't set height for Plotly charts: on Dashboard page chart fits it's container in widget, and on Query page it has fixed height (IIRC, `500px`) and width `100%`.

Reply:
So the default height `500px` comes from Plot.ly library itself?

I implemented `100%` height for Vega charts everywhere.